FSQLocationBroker/FSQLocationBroker.m

```diff
@@ -516,13 +516,13 @@ - (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray
     BOOL isBackgrounded = applicationIsBackgrounded();
     CLLocation *mostRecentLocation = [locations firstObject];
     for (CLLocation *location in locations) {
-        if ([mostRecentLocation.timestamp laterDate:location.timestamp] == location.timestamp) {
+        if ([mostRecentLocation.timestamp compare:location.timestamp] == NSOrderedAscending) {
```

Comment:
So after reading all of these things it sounds like the case we are trying to handle is when `locationManager:didUpdateLocations` contains an array when `locations.count == 0` is that correct? 

I am not sure that will ever happen, but wouldn't it make more sense to just exit the function if it is called with an empty locations array?